src/lib.rs

```diff
@@ -513,10 +535,61 @@ impl Octocrab {
         self.execute(request).await
     }
 
+    #[cfg(not(feature = "rate-limiting"))]
     /// Execute the given `request` octocrab's Client.
     pub async fn execute(&self, request: reqwest::RequestBuilder) -> Result<reqwest::Response> {
         request.send().await.context(error::Http)
     }
+
+    #[cfg(feature = "rate-limiting")]
+    /// Execute the given `request` octocrab's Client.
+    pub async fn execute(&self, request: reqwest::RequestBuilder) -> Result<reqwest::Response> {
+        if let Some(delay) = {
+            let mut limiter = self.rate_limiter
+                .as_ref()
+                .lock()
+                .expect("Poisoned rate limiter mutex");
+            limiter.request_delay()
+        } {
+            delay.await;
+        }
+
+        {
+            self.rate_limiter
+                .as_ref()
+                .lock()
+                .expect("Poisoned rate limiter mutex")
+                .register_request();
+        }
+
+        let request_cloned = request.try_clone().unwrap(); // we never use streams in bodies
+        let response = request.send().await.context(error::Http);
+
+        let delay = {
+            let mut limiter = self.rate_limiter
+                .as_ref()
+                .lock()
+                .expect("Poisoned rate limiter mutex");
+            limiter.register_response(&response);
+            if let RateLimiterState::RateLimited(reset) = &limiter.state {
+                (*reset - Utc::now()).to_std().ok()
+            } else {
+                None
+            }
+        };
+
+        if let (Ok(res), Some(delay)) = (&response, &delay) {
+            if res.status() == reqwest::StatusCode::FORBIDDEN {
```

Comment:
This should have the same behaviour as octokit's retry plugin, which doesn't retry on Forbidden. https://github.com/octokit/plugin-retry.js

Reply:
Got it. I'll take a look.